pkg/frontend/querymiddleware/codec.go

```diff
@@ -180,6 +180,7 @@ func (prometheusCodec) MergeResponse(responses ...Response) (Response, error) {
 	}
 
 	promResponses := make([]*PrometheusResponse, 0, len(responses))
+	promWarnings := make([]string, 0)
```

Comment:
Could we do this instead? This would avoid the need to set `promWarnings` to `nil` later if it's empty.

```suggestion
	var promWarnings []string
```